web/src/app/workplace/components/taskdetails-general/general-fields.component.ts

```diff
@@ -102,14 +102,15 @@ export class TaskdetailsGeneralFieldsComponent implements OnInit, OnChanges, OnD
     });
   }
 
+  // TODO: this is currently called for every selected task and is only necessary when we switch the workbasket -> can be optimized.
   private getClassificationByDomain() {
     this.requestInProgress = true;
     this.classificationService
-      .getClassificationsByDomain(this.domainService.getSelectedDomainValue())
-      .then((classificationPagingList) => {
+      .getClassifications({ domain: [this.task.workbasketSummary.domain] })
+      .subscribe((classificationPagingList) => {
```

Comment:
I think the unsubscription is missing ( `.pipe(takeUntil(this.$destroy))` like in the other components)

Reply:
I only reformatted this. But I can do that